src/Domain/DomainHelpers.cpp

```diff
@@ -261,6 +261,66 @@ obtain_correspondence_between_blocks(
                                               nhbr_align);
 }
 
+template <size_t VolumeDim>
+std::vector<std::array<size_t, two_to_the(VolumeDim)>> corners_from_two_maps(
+    const std::unique_ptr<domain::CoordinateMapBase<
+        Frame::Logical, Frame::Inertial, VolumeDim>>& map1,
+    const std::unique_ptr<
+        domain::CoordinateMapBase<Frame::Logical, Frame::Inertial, VolumeDim>>&
+        map2) noexcept {
+  std::array<size_t, two_to_the(VolumeDim)> corners_for_block1;
+  std::array<size_t, two_to_the(VolumeDim)> corners_for_block2;
+  for (VolumeCornerIterator<VolumeDim> vci{}; vci; ++vci) {
+    gsl::at(corners_for_block1, vci.local_corner_number()) =
+        vci.local_corner_number();
+  }
+  // Fill corners_for_block2 based off which corners are shared with map1.
+  for (VolumeCornerIterator<VolumeDim> vci_map2{}; vci_map2; ++vci_map2) {
+    const auto mapped_coords2 =
+        (*map2)(tnsr::I<double, VolumeDim, Frame::Logical>{
+            vci_map2.coords_of_corner()});
+    double num_unshared_corners = 0;
+    // Check each corner of map1 to see if it shares any corners with map2.
+    for (VolumeCornerIterator<VolumeDim> vci_map1{}; vci_map1; ++vci_map1) {
+      const auto mapped_coords1 =
+          (*map1)(tnsr::I<double, VolumeDim, Frame::Logical>{
+              vci_map1.coords_of_corner()});
+      double max_separation = 0.0;
+      for (size_t j = 0; j < VolumeDim; j++) {
+        max_separation =
+            std::max(max_separation,
+                     std::abs(mapped_coords2.get(j) - mapped_coords1.get(j)));
+      }
+      // If true, then the mapped_coords lie on top of one another.
+      // This corner of map2 is assigned the same number as that of map1.
+      if (max_separation < 1.0e-6) {
```

Comment:
Why `1.0e-6`? Could this be passed in as an argument to the function? Could `equal_within_roundoff` be used here? My preferred setup, if possible, would be taking `epsilon` and `scale` as arguments, then passing them to `equal_within_roundoff` here. It seems like ideally the arguments would be specifiable in `set_internal_boundaries` so each domain creator can adjust them as necessary.

Reply:
I'm not sure about changing this, because so far we've used analytic maps that are constructed to agree at the boundaries. If `max_separation` is above any small number, the maps are most likely disconnected, and if they are connected, they share an analytic map at the boundary so the `max_separation` is likely very small. I chose `1.0e-6` because so far all separations in the domains I've tested have been clearly on one side of this bound or the other. To have an argument be passed in suggests the map connectivity will be "fuzzy" where they're only connected up to some tolerance.

An alternative I would suggest, if desired, is an additional function that finds the `max_separation` in a domain where the maps are properly connected, that would be a good diagnostic in some cases.

Reply:
I guess my question is more if I rescale the entire domain radially by 1.0e8 then `1.0e-6` may no longer be large enough to check agreement. It seems like the separation is, even for analytic maps, a relative quantity compared to some measure of the location of the boundary/size of the domain.  E.g. if we took the BBH domain and just made the holes be 1e10 solar masses then the separation between domains should probably be scaled according to that size. I'm not sure if we'll encounter this in practice. I'm just asking to see what you're thoughts are :)

Reply:
I agree - ideally the threshold is some size relative to the size of the maps and I've assumed that in practice all maps have a size of order ~1 at least to within a couple of orders of magnitude. I can add a comment and/or error message that discusses this, because I think if we do encounter this it will be in special cases only. :+1:

Reply:
Okay, sounds good!